### PR TITLE
Add deletion protection to resource_compute_instance

### LINKED ISSUE #956: Request: add support for "accidental VM deletion" in "google_compute_instance"
Hi,

It could be great to get support for the "accidental VM deletion" for the "google_compute_instance" - now that the feature got into GA.

And thank you for an excellent provider! :)

### PR DESCRIPTION
**Pull request by nickjacques**
- [x] Added `deletion_protection` boolean parameter for `google_compute_instance` resources (in `resource_compute_instance.go`).
  - Parameter is optional, default value is false
- [x] Updated documentation to describe functionality
- [x] Add unit tests to confirm functionality

This resolves #956.

### CODE REVIEW

**Comment by nickjacques** [comment]
Added unit tests:
- Added implicit false test to `TestAccComputeInstance_basic1`
- Created explicit false test with `TestAccComputeInstance_deletionProtectionExplicitFalse`
- Created explicit true test with `TestAccComputeInstance_deletionProtectionExplicitTrue`

@danawillow I think this is ready for review!

```bash
$ make testacc TEST=./google TESTARGS='-run=TestAccComputeInstance_basic1'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google -v -run=TestAccComputeInstance_basic1 -timeout 120m
=== RUN   TestAccComputeInstance_basic1
--- PASS: TestAccComputeInstance_basic1 (59.16s)
PASS
ok      github.com/terraform-providers/terraform-provider-google/google 59.188s
```

```bash
$ make testacc TEST=./google TESTARGS='-run=TestAccComputeInstance_deletionProtectionExplicitFalse'                                                      
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google -v -run=TestAccComputeInstance_deletionProtectionExplicitFalse -timeout 120m
=== RUN   TestAccComputeInstance_deletionProtectionExplicitFalse
--- PASS: TestAccComputeInstance_deletionProtectionExplicitFalse (179.77s)
PASS
ok      github.com/terraform-providers/terraform-provider-google/google 179.803s
```

```bash
$ make testacc TEST=./google TESTARGS='-run=TestAccComputeInstance_deletionProtectionExplicitTrueAndUpdateFalse'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google -v -run=TestAccComputeInstance_deletionProtectionExplicitTrueAndUpdateFalse -timeout 120m
=== RUN   TestAccComputeInstance_deletionProtectionExplicitTrueAndUpdateFalse
--- PASS: TestAccComputeInstance_deletionProtectionExplicitTrueAndUpdateFalse (81.45s)
PASS
ok      github.com/terraform-providers/terraform-provider-google/google 81.485s
```

File: google/resource_compute_instance_test.go [hunk 1]
```diff
@@ -951,6 +955,61 @@ func TestAccComputeInstance_minCpuPlatform(t *testing.T) {
...
+		CheckDestroy: testAccCheckComputeInstanceDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeInstance_basic_deletionProtectionTrue(instanceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeInstanceExists(
+						"google_compute_instance.foobar", &instance),
+					testAccCheckComputeInstanceHasConfiguredDeletionProtection(&instance, true),
+				),
+			},
```
Comment (ndmckinley) [inline google/resource_compute_instance_test.go]: Could you toss an `ImportStateVerify` test step here between step 0 and step 1?  That way we can be sure that this works on import.  I'm sure it does, it's just something we're trying to do lately.

**Comment by ndmckinley** [comment]
This looks great!  Thanks for the contribution.

File: google/resource_compute_instance_test.go (hunk 1 above)
Comment (nickjacques) [reply google/resource_compute_instance_test.go]: Sure! Updating commit incoming.

**Comment by ndmckinley** [comment]
Great!  Running acceptance tests now.

**Comment by ndmckinley** [comment]
Tests pass, great.  :)

**Pull request merged**